static/js/filter.js

```diff
@@ -382,13 +382,38 @@ Filter.prototype = {
     },
 
     can_mark_messages_read: function () {
-        return _.every(this._operators, function (elem) {
-            return (_.contains(['stream', 'topic', 'pm-with'], elem.operator)
-                || elem.operator === 'is' && elem.operand === 'private'
```

Comment:
We intend to switch `is:mentioned` to work like `is:private`; probably that'll merge before this but FYI.